Add a VS Code dev container

brobeson opened the pull request:
- I added the files for a VS Code dev container.
- I committed the incorrect *projects/README.md* file. See #79 for the bug this exposes.

closes #69

brobeson committed:
Add a VS Code dev container

- I added the docker file for the container.
- I added devcontainer.json.
- I added build documentation.

closes #69

brobeson committed:
Update projects/readme

There is a bug which makes externpro think we have no patch files.
See #79. This commits the updated readme, that records no patches.
Even though the readme is incorrect, this should prevent confusion
among developers until I fix #79.

bnhhughes commented on .devcontainer/Dockerfile:
```diff
@@ -0,0 +1,63 @@
+  yum install --assumeyes \
+    epel-release \
+    # This repository gives us more modern versions of some software, such as Git.
+    https://packages.endpoint.com/rhel/7/os/x86_64/endpoint-repo-1.7-1.x86_64.rpm; \
```
Thanks for this, by the way - I was still using an old version of git on my CentOS VM until you mentioned this one includes a newer version.

swflb commented on .devcontainer/Dockerfile:
```diff
@@ -0,0 +1,63 @@
+    make \
+    wget; \
+  # yum --assumeyes clean all; \
+  ln -s cmake3 /usr/bin/cmake; \
```
Do you also want to add `ln -s ccmake3 /usr/bin/ccmake; `?

brobeson replied:
Sure, I overlooked that one.

swflb commented on .devcontainer/Dockerfile:
```diff
@@ -0,0 +1,63 @@
+  ln -s cmake3 /usr/bin/cmake; \
+  ln -s ctest3 /usr/bin/ctest; \
+  ln -s cpack3 /usr/bin/cpack; \
+  yum install --assumeyes \
```
Why do 2 separate install steps? I think you should combine this step with line 14.

brobeson replied:
I prefer to have related functionality close together.  Hence creating the symlinks soon after installing CMake. Also, the first yum install is just necessary tools; the second is dependencies for specific ovsrpro projects. I'll change it, though.

swflb commented on .devcontainer/Dockerfile:
```diff
@@ -0,0 +1,63 @@
+    libxcb-devel \
+    libxkbcommon-devel \
+    openssl-devel \
+    which \
```
I am surprised `which` is not installed by default on a RHEL or CentOS system.

brobeson replied:
So was I the first time that I encountered that missing gem.

swflb commented on .devcontainer/Dockerfile:
```diff
@@ -0,0 +1,63 @@
+    autoconf \
+    automake \
+    java-11-openjdk-devel \
+    libtool; \
```
do you not need these

- ncurses-devel
- ninja-build
- glibs
- openssl
- dbus
- libXScrnSaver 
- libXtst
- pciutils
- EGL
- javac
I thought those were all necessary. but perhaps some of those are necessary only for developing applications that depend on ovsrpro. ?

brobeson replied:
Nope. ncurses-devel, openssl, dbus, libXScrnSave, libXtst, pciutils, and EGL are not needed. Building ovsrpro does not work with Ninja, see #72.

glibs-devel might have solved the Qt build error, but I didn't know about that. I tried glibc-devel and glib-devel, neither of which worked. So I just removed `-glib` in #71

swflb commented:
@brobeson thanks for the explanations

swflb commented:
@brobeson I just remembered, if we are going to be using VSCode, we might want this added to the .gitignore file
```
*vscode
```

brobeson commented:
> @brobeson I just remembered, if we are going to be using VSCode, we might want this added to the .gitignore file
> 
> ```
> *vscode
> ```

This is one that I think everyone should add to their global Git ignore file.

swflb commented:
`vim` might be handy to install. ?

brobeson commented:
> @brobeson thanks for the explanations

You're welcome. The changes will take awhile to push up. I need to rebuild the container, then rebuild ovsrpro to ensure it all still works.

brobeson committed:
Clean up the Dockerfile

- I combined the yum install commands into one command.
- I added a symlink for ccmake.
- This is response to comments made on pull request #80.

brobeson commented:
> `vim` might be handy to install. ?

The container's use case is with VS Code, so I don't see a need for an editor.

brobeson committed:
Ignore the .vscode directory in Git
